exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/partitionsender/PartitionerTemplate.java

```diff
@@ -161,8 +161,11 @@ public OperatorStats getStats() {
    * @param schemaChanged  true if the schema has changed
    */
   @Override
-  public void flushOutgoingBatches(boolean isLastBatch, boolean schemaChanged) throws IOException {
+  public void flushOutgoingBatches(boolean isLastBatch, boolean schemaChanged) throws IOException, InterruptedException {
     for (OutgoingRecordBatch batch : outgoingBatches) {
+      if (Thread.interrupted()) {
+        throw new InterruptedException();
```

Comment:
Since we are checking for interrupts here already could we remove `Thread.currentThread().isInterrupted()` in the flush(boolean schemaChanged) method?

Reply:
I'll revert back throwing 'InterruptedException' to avoid mishandling of the last batch.